src/prettyprint.lua

```diff
@@ -0,0 +1,229 @@
+	---@param header string Header to display
+	---@param failed number
+	---@param passed number
+	---@param total number
+	---@return string rendered Rendered display
+	local function createSummaryInfo(header, failed, passed, total)
+		local stringTable = {}
+		if failed > 0 then
+			tablex.push(
+				stringTable,
+				COLOURS.TESTS_FAILED(("%d failed"):format(failed))
+			)
+		end
+
+		if passed > 0 then
+			tablex.push(
+				stringTable,
+				COLOURS.TESTS_PASSED(("%d passed"):format(passed))
+			)
+		end
+
+		tablex.push(stringTable, ("%d total"):format(total))
+		return ("%s %s"):format(
+			COLOURS.BOLD(header),
+			table.concat(stringTable, ", ")
+		)
+	end
+
+	print(
+		createSummaryInfo(
+			"Test Suites:",
+			failedSuites,
+			passedSuites,
+			totalSuites
+		)
+	)
+
+	print(createSummaryInfo("Tests:", failedTests, passedTests, totalTests))
+end
+
+--- Prints detailed error reports about the tests that failed.
+---@param results lest.TestSuiteResults
+local function printTestErrors(results)
+	---@type table<number, {displayName: string, result: lest.TestResult}>
+	local failedTests = {}
+
+	---@type table<number, string>
+	local parentNames = {}
+	for _, testSuite in ipairs(results) do
+		if not testSuite.passed then
+			traverseNodes(testSuite, function(node)
+				if node.type == NodeType.Describe then
+					if node.pass then
+						-- Skip traversing this branch as it doesn't have anything that failed.
+						return false
+					end
+
+					tablex.push(parentNames, node.name)
+				elseif node.type == NodeType.Test and not node.pass then
+					tablex.push(failedTests, {
+						displayName = table.concat(parentNames, " › ")
+							.. " › "
+							.. node.name,
+						result = node,
+					})
+				end
+
+				return true
+			end, function()
+				tablex.pop(parentNames)
+			end)
+		end
+	end
+
+	for _, failedTest in ipairs(failedTests) do
+		print(
+			COLOURS.ERROR_HEADER(
+				string.format(" %s %s\n", ERROR_SYMBOL, failedTest.displayName)
+			)
+		)
+
+		if
+			type(failedTest.result.error) == "table"
+			and failedTest.result.error.signature
+		then
+			print("   " .. failedTest.result.error.signature .. "\n")
+			print(
+				"   " .. COLOURS.FAIL(failedTest.result.error.message) .. "\n"
+			)
+		else
+			print("   " .. prettyValue(failedTest.result.error) .. "\n")
+		end
+	end
+end
+
+--- Prints detailed reports about the test suites in the console.
+---@param results lest.TestSuiteResults
+local function printDetailedReports(results)
+	--- Replicates the Jest test header output
+	---@param testSuite lest.TestSuiteResults
+	local function printHeader(testSuite)
+		local pathComponents = {}
+		for component in testSuite.name:gmatch("([^%/]*)") do
+			tablex.push(pathComponents, component)
+		end
+
+		local renderedName = COLOURS.DIMMED(
+			table.concat(pathComponents, "/", 1, #pathComponents - 1) .. "/"
+		) .. COLOURS.FILENAME(pathComponents[#pathComponents])
```

Comment:
Why are you splitting every element just to recolour the last?

Can you not simply split on the _last_ instance of `/`?

Reply:
I dont.. think so with how lua patterns work.

Reply:
https://stackoverflow.com/questions/14554193/last-index-of-character-in-string